packages/medusa/src/services/event-bus.ts

```diff
@@ -99,22 +98,52 @@ export default class EventBusService {
     const cloned = new EventBusService(
       {
         manager: transactionManager,
-        stagedJobRepository: this.stagedJobRepository_,
         jobSchedulerService: this.jobSchedulerService_,
         logger: this.logger_,
         redisClient: this.redisClient_,
         redisSubscriber: this.redisSubscriber_,
+        cacheService: this.cacheService_,
       },
       this.config_,
       false
     )
 
-    cloned.transactionManager_ = transactionManager
     cloned.queue_ = this.queue_
 
     return cloned
   }
 
+  /**
+   * Processes jobs in the queue.
+   * @param job The job object
+   * @return resolves to the results of the subscriber calls.
+   */
+  worker_ = async <T>(job: {
+    data: { eventName: string; data: T }
+  }): Promise<unknown[]> => {
+    const { eventName, data } = job.data
+    const eventObservers = this.observers_.get(eventName) || []
+    const wildcardObservers = this.observers_.get("*") || []
+
+    const observers = eventObservers.concat(wildcardObservers)
+
+    this.logger_.info(
+      `Processing ${eventName} which has ${eventObservers.length} subscribers`
+    )
+
+    return await Promise.all(
+      observers.map(async (subscriber) => {
+        return subscriber(data, eventName).catch((err) => {
```

Comment:
todo: As we discussed, since the event act as an independent transaction flow, we should create a uniq cache id and pass it as part of a context argument to the subscriber. Then we should add `.then` to call the processEventsCached and in the `.catch` we should call the invalidate